regression/validate-trace-xml-schema/check.py

```diff
@@ -0,0 +1,166 @@
+import os
+import subprocess
+import sys
+import tempfile
+import glob
+
+this_script_dir = os.path.abspath(os.path.dirname(__file__))
+
+TraceXsdSpec = os.path.abspath(os.path.join(this_script_dir, '..', '..', 'doc/assets/xml_spec.xsd'))
+
+test_base_dir = os.path.abspath(os.path.join(this_script_dir, '..', 'cbmc'))
+
+# some tests in the cbmc suite don't work for the trace checks for one reason or another
+ExcludedTests = list(map(lambda s: os.path.join(test_base_dir, s), [
+    # these tests except input from stdin
+    'json-interface1/test_wrong_option.desc',
+    'json-interface1/test.desc',
+    'json-interface1/test_wrong_flag.desc',
+    'xml-interface1/test_wrong_option.desc',
+    'xml-interface1/test.desc',
+    'xml-interface1/test_wrong_flag.desc',
+    # these want --show-goto-functions instead of producing a trace
+    'destructors/compound_literal.desc',
+    'destructors/enter_lexical_block.desc',
+    'reachability-slice-interproc2/test.desc',
+    # this one wants show-properties instead producing a trace
+    'show_properties1/test.desc',
+    # program-only instead of trace
+    'vla1/program-only.desc',
+    'Quantifiers-simplify/simplify_not_forall.desc',
+    # this one doesn't work for me locally at all
+    # 'integer-assignments1/test.desc',
```

Comment:
>  Huh? You mean the test fails locally? Perhaps we could raise an issue to look into that?

:question: outstanding, this needs to be addressed IMO

Reply:
To be explicit about what needs addressing:
 - shouldn't check in "me" since no way of knowing who "me" is as soon as this line gets moved. 

Further, I'd like to understand what you mean "doesn't work locally at all" means!